drivers: spi: Add driver support for TI MSPM0 SPI module

santhosh-c-c opened the pull request:
**Add SPI driver support for TI’s MSPM0 microcontroller series.**

This driver implements SPI master mode with configurable parameters including frame size (4–16 bits), clock polarity, clock phase, and bit order. It integrates with Zephyr’s SPI subsystem and provides standard API compliance.

**Signed-off-by:**
Santhosh Charles [santhosh@linumiz.com](mailto:santhosh@linumiz.com)

santhosh-c-c committed:
dts: bindings: spi: add DT bindings for TI MSPM0 SPI module

Add devicetree bindings for the TI MSPM0 SPI module.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>
Signed-off-by: Jackson Farley <j-farley@ti.com>

santhosh-c-c committed:
drivers: spi: add driver support for TI MSPM0 SPI module

Add support for the SPI module on TI’s MSPM0 MCUs.
The driver supports master mode transfers with configurable
frame size (4–16 bits), clock polarity/phase, bit order.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>
Signed-off-by: Jackson Farley <j-farley@ti.com>
Co-authored-by: Hans Binderup <habi@bang-olufsen.dk>

santhosh-c-c committed:
dts: arm: ti: mspm0: add SPI nodes for TI MSPM0

Add SPI peripheral nodes to the MSPM0 family dtsi to
enable spi support for the G and L series.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>
Signed-off-by: Jackson Farley <j-farley@ti.com>

tbursztyka reviewed (dismissed):
Please, use @hansbinderup's original commit as first, do not modify it.

Then add your modifications (ti,clk-div and few details) in further commits.

Also: do not add your copyright unless significant changes (like: adding interrupt mode support for instance).

santhosh-c-c committed:
manifest: update hal_ti to add mspm0l222x SPI pin properties

Update hal_ti module revision to add the input-enable
property for all SPI POCI pins for the mspm0l222x series.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

ssekar15 commented on drivers/spi/spi_mspm0.c:
```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (c) 2024 Bang & Olufsen A/S, Denmark
```
2025

parthitce commented on drivers/spi/spi_mspm0.c:
```diff
@@ -0,0 +1,353 @@
+	uint8_t dfs;
+
+	if (!tx_bufs && !rx_bufs) {
+		return 0;
```
EINVAL?

parthitce commented on drivers/spi/spi_mspm0.c:
```diff
@@ -0,0 +1,353 @@
+	/* Transmit dummy frame when no TX data is provided */
+	uint32_t tx_frame = 0;
+
+	k_sem_reset(&data->spi_idle);
```
This is not protecting anything and MULTITHREADING will fail. Anyways, move to spi_context

parthitce commented on drivers/spi/spi_mspm0.c:
```diff
@@ -0,0 +1,353 @@
+	const struct spi_mspm0_config *config = dev->config;
+	struct spi_mspm0_data *data = dev->data;
+
+	/* Transmit dummy frame when no TX data is provided */
```
Any specific reason?

santhosh-c-c commented:
@etienne-lms @gautierg-st @nika-nordic @lsd-he360 
Ping for Review, Thanks!

santhosh-c-c committed:
boards: ti: lp_mspm0g3507: enable SPI support

Enable SPI peripheral and add spi to the list of supported
hardware in board YAML.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c committed:
boards: ti: lp_mspm0g3519: enable SPI support

Enable SPI peripheral and add spi to the list of supported
hardware in board YAML.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c committed:
boards: ti: lp_mspm0l2228: enable SPI support

Enable SPI peripheral and add spi to the list of supported
hardware in board YAML.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c committed:
tests: drivers: spi: add MSPM0G3507 to spi_loopback test

Add board overlay and config for lp_mspm0g3507 to enable
the spi_loopback test suite using internal loopback mode.

Signed-off-by: Santhosh Charles <santhosh@linumiz.com>

santhosh-c-c commented:
@teburd @tmon-nordic @str4t0m
Ping for Review, Thanks!

parthitce commented on drivers/spi/spi_mspm0.c:
```diff
@@ -0,0 +1,337 @@
+	SPI_Regs *regs;
+	int ret;
+
+	DEVICE_MMIO_MAP(dev, K_MEM_CACHE_NONE);
```
DEVICE_MMIO_MAP emits empty do - while when MMIO is not in RAM. IMO this line can be dropped.

parthitce reviewed (dismissed):
LGTM

ssekar15 commented on boards/ti/lp_mspm0g3507/lp_mspm0g3507.dts:
```diff
@@ -133,3 +133,23 @@
+	status = "okay";
+	pinctrl-0 = <&spi0_sclk_pb18 &spi0_pico_pb17 &spi0_poci_pb19 &spi0_cs0_pa8>;
+	pinctrl-names = "default";
+	cs-gpios = <&gpiob 6 GPIO_ACTIVE_LOW>;
```
gpioa 8
